test/model/test/common/util.cpp

```diff
@@ -63,3 +67,49 @@ create_tmp_file(const char *dir, const char *prefix, const char *suffix)
 
     return std::string(buf);
 }
+
+/*
+ * verify_using_debug_log --
+ *     Verify the database using the debug log. Try both the regular and the JSON version.
+ */
+void
+verify_using_debug_log(TEST_OPTS *opts, const char *home, bool test_failing)
+{
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+
+    testutil_wiredtiger_open(
+      opts, home, "log=(enabled,file_max=10M,remove=false)", nullptr, &conn, false, false);
+    testutil_check(conn->open_session(conn, nullptr, nullptr, &session));
+    std::vector<std::string> tables = model::wt_list_tables(conn);
+
+    /* Verify using the debug log. */
+    model::kv_database db_from_debug_log;
+    model::debug_log_parser::from_debug_log(db_from_debug_log, conn);
+    for (auto &t : tables)
+        testutil_assert(db_from_debug_log.table(t.c_str())->verify_noexcept(conn));
+
+    /* Print the debug log to JSON. */
+    std::string tmp_json = create_tmp_file(home, "debug-log-", ".json");
+    wt_print_debug_log(conn, tmp_json.c_str());
```

Comment:
How do you know that the WiredTiger debug log is the same at both new line 88 and new line 94? This code assumes that is true. Perhaps you should call wt_print_debug_log() all the following places: before new line 88, at new line 94, and after new line 101.  Then compare all the JSON files to make sure they are the same.